client/src/components/PacmanGame/gameCore.js

```diff
@@ -45,35 +39,39 @@ export const getGhosts = () => [[23, 5], [19, 1], [1, 19], [1, 13]];
 export const getPacman = () => [[43, 5]];
 
 export const entityApplier = (gridState, entityLocations, entityCode) => {
-  for (const [x, y] of entityLocations) {
-    gridState[x][y] = entityCode;
-  }
-  // return gridState;
+  const newState = [...gridState];
+  // console.log(entityLocation);
+  entityLocations.map(([x, y]) => {
```

Comment:
Why not reduce? with initial state as gridState?

Reply:
Refactored with `.reduce`.